File: src/api/routers/memory.py
"""记忆路由(模块 11 T3):GET /memory / DELETE /memory/{key}。

复用 06 的 PostgresStore(namespace ("memory", "default"),与 REPL /memory 同语义)。
"""
from fastapi import APIRouter

from settings.config import get_settings
from settings.db.store import get_store

router = APIRouter(prefix="/memory", tags=["memory"])


def _store():
    return get_store(get_settings().database_url)


@router.get("")
def list_memory():
    items = _store().search(("memory", "default"), query=None, limit=100)
    items = sorted(items, key=lambda it: it.value.get("created_at", ""), reverse=True)
    return {
        "items": [
            {
                "key": it.key,
                "content": it.value.get("content", ""),
                "source": it.value.get("source", ""),
                "created_at": it.value.get("created_at", ""),
            }
            for it in items
        ]
    }


@router.delete("/{key}")
def delete_memory(key: str):
    _store().delete(("memory", "default"), key)
    return {"ok": True, "key": key}

File: dev/front/src/types/memory.ts
/**
 * Memory 资源类型镜像(唯一事实源:API-CONTRACT.md §2.4)。
 */

/** GET /memory 单条长期记忆。 */
export interface MemoryItem {
  /** Store 主键。 */
  key: string
  /** 一句话原子事实;缺失时为 ""。 */
  content: string
  /** "explicit"(用户明示)| "confirmed"(自动确认);缺失时为 ""。 */
  source: 'explicit' | 'confirmed' | ''
  /** ISO 8601;缺失时为 ""。 */
  created_at: string
}

/** GET /memory 响应(按 created_at 降序,固定 limit=100)。 */
export interface MemoryListResponse {
  items: MemoryItem[]
}

/** DELETE /memory/{key} 响应;store.delete() 对不存在的 key 幂等,也返回 200。 */
export interface MemoryDeleteResult {
  ok: boolean
  key: string
}